src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -1860,6 +1874,7 @@ protected override string GetSuspensionPointDisplayName(SyntaxNode node, EditKin
                 _kind = kind;
                 _span = span;
                 _match = match;
+                _isTopLevelEdit = isTopLevelEdit;
```

Comment:
>_isTopLevelEdit [](start = 16, length = 15)

I'm not seeing why is this flag necessary. Wouldn't the syntax kinds that are not found in local functions be just ignored?

Reply:
Yeah, I think this can be removed now, just haven't gone back to check yet. This is from previous attempts where the method body itself was classified, and I needed a way to ignore more node types.

Reply:
So this is still an issue because the EditClassifier is now seeing edits for nodes that StatementSyntaxComparer cares about, but TopSyntaxComparer doesn't, and it was written for the latter I guess.

1. Change `_isTopLevelEdit` to `_throwOnUnexpectedNode` and have it just control the `throw ExceptionUtilities.UnexpectedValue` calls
2. Check `HasLabel` before throwing unexpected value
3. Add all of the missing syntax kinds to the EditClassifier and just return without doing anything
4. Leave as is

Thoughts? Or have I missed something about when to class the EditClassifier?

Reply:
I'd check `HasLabel` _before_ calling the classifier for a specific edit.